kratos/utilities/math_utils.h

```diff
@@ -1636,51 +1650,78 @@ class MathUtils
+                    aux_A(index2, i) = rEigenValuesMatrix(index2, i) + s * (rEigenValuesMatrix(index1, i)- teta * rEigenValuesMatrix(index2, i));
+                    aux_A(i, index2) = rEigenValuesMatrix(index2, i) + s * (rEigenValuesMatrix(index1, i)- teta * rEigenValuesMatrix(index2, i));
+                    aux_A(index1, i) = rEigenValuesMatrix(index1, i) - s * (rEigenValuesMatrix(index2, i) + teta * rEigenValuesMatrix(index1, i));
+                    aux_A(i, index1) = rEigenValuesMatrix(index1, i) - s * (rEigenValuesMatrix(index2, i) + teta * rEigenValuesMatrix(index1, i));
                 }
             }
 
-            temp_mat = aux_A;
+            noalias(rEigenValuesMatrix) = aux_A;
 
             // Calculation of the eigeneigen vector matrix V
-            rotation_matrix = identity_matrix;
+            noalias(rotation_matrix) = identity_matrix;
             rotation_matrix(index2, index1) = -s;
             rotation_matrix(index1, index2) =  s;
             rotation_matrix(index1, index1) =  c;
             rotation_matrix(index2, index2) =  c;
 
-            aux_V_matrix = ZeroMatrix(TDim, TDim);
+            noalias(aux_V_matrix) = ZeroMatrix(size, size);
 
-            for(IndexType i = 0; i < TDim; ++i) {
-                for(IndexType j = 0; j < TDim; ++j) {
-                    for(IndexType k = 0; k < TDim; ++k) {
-                        aux_V_matrix(i, j) += V_matrix(i, k) * rotation_matrix(k, j);
+            for(IndexType i = 0; i < size; ++i) {
+                for(IndexType j = 0; j < size; ++j) {
+                    for(IndexType k = 0; k < size; ++k) {
+                        aux_V_matrix(i, j) += rEigenVectorsMatrix(i, k) * rotation_matrix(k, j);
                     }
                 }
             }
-            V_matrix = aux_V_matrix;
+            noalias(rEigenVectorsMatrix) = aux_V_matrix;
         }
 
         KRATOS_WARNING_IF("MathUtils::EigenSystem", !is_converged) << "Spectral decomposition not converged " << std::endl;
 
+        return is_converged;
+    }
+
+    /**
+     * @brief Calculates the eigenvectors and eigenvalues of given symmetric TDimxTDim matrix
+     * @details The eigenvectors and eigenvalues are calculated using the iterative Gauss-Seidel-method. The resulting decomposition is L'DL
+     * @note See https://en.wikipedia.org/wiki/Gauss%E2%80%93Seidel_method
+     * @param A The given symmetric matrix the eigenvectors are to be calculated.
+     * @param rEigenVectorsMatrix The result matrix (will be overwritten with the eigenvectors)
+     * @param rEigenValuesMatrix The result diagonal matrix with the eigenvalues
+     * @param Tolerance The largest value considered to be zero
+     * @param MaxIterations Maximum number of iterations
+     * @tparam TDim The size of the bounded matrix
+     * @warning This method is deprecated. Will be removed soon
+     */
+    template<SizeType TDim>
+    KRATOS_DEPRECATED_MESSAGE("Please use GaussSeidelEigenSystem() instead")
```

Comment:
maybe add in the note that the output is in transpose in the new function

Reply:
OK